### PR TITLE
Emit suggestions and an explanation when CBMC runs out of memory

### LINKED ISSUE #2715: CBMC failed with status 137
With Kani version 0.35.0 some of our previously working proofs are now failing with the message `CBMC failed with status 137`.

I don't have the ability right now to provide the rust code.

### PR DESCRIPTION
**Pull request by JustusAdam**
Improves the error reported to the user when the system kills CBMC to reclaim memory.

Resolves #2715 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

**Review by celinval — APPROVED** [review]
That's great! Thanks.

As a future improvement we should create a wiki with different ways of improving the memory consumption and link it here.

**Comment by JustusAdam** [comment]
> That's great! Thanks.
> 
> As a future improvement we should create a wiki with different ways of improving the memory consumption and link it here.

I was thinking the same thing.

**Comment by JustusAdam** [comment]
@celinval Since this is a benign change it's probably all right if I don't wait for a second review?

**Comment by adpaco-aws** [comment]
> @celinval Since this is a benign change it's probably all right if I don't wait for a second review?

You don't need to wait for a second review. I'll merge it for you this time 😛

**Pull request merged**
